src/tapedfunction.jl

```diff
@@ -219,8 +221,9 @@ function (instr::Instruction{F})(tf::TapedFunction, callback=nothing) where F
         output = if is_primitive(func, inputs...)
             func(inputs...)
         else
-            tf_inner = TapedFunction(func, inputs..., cache=true)
-            tf_inner(inputs...; callback=callback)
+            tf_inner = get!(tf.subtapes, instr, TapedFunction(func, inputs..., cache=true))
```

Comment:
This will always construct a `TapedFunction` object, regardless of whether `tf.subtapes[instr]` exists or not. Maybe instead we could use
```suggestion
            tf_inner = get!(tf.subtapes, instr) do
                TapedFunction(func, inputs...; cache=true)
            end
```